components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -156,12 +137,27 @@ func (g *gpsrtk) receiveAndWriteCorrectionData() {
 	defer g.activeBackgroundWorkers.Done()
 	defer g.closeCorrectionWriter()
 
-	err := g.connectToNTRIP()
+	select {
+	case <-g.cancelCtx.Done():
+		g.err.Set(errors.New("context canceled"))
+		return
+	default:
+	}
+
+	err := g.connectAndParseSourceTable()
 	if err != nil {
 		g.err.Set(err)
-		g.logger.Error("unable to connect to NTRIP stream! Giving up on RTK messages")
+		g.logger.Error("unable to parse source table! Giving up on RTK messages")
```

Comment:
the "Giving up on RTK messages" part of the message doesn't make much sense. It might be better to just say `"Encountered error while parsing source table", err ` and return what error connectAndParseSourceTable is returning